odoo/addons/test_new_api/tests/test_domain.py

```diff
@@ -39,3 +41,69 @@ def test_00_test_bool_undefined(self):
 
             self.assertEqual(eq_1+eq_2, all_bool, 'True + False != all')
             self.assertEqual(neq_1+neq_2, all_bool, 'not True + not False != all')
+
+    def test_empty_char(self):
+        EmptyChar = self.env['test_new_api.empty_char']
+        EmptyChar.create([
+            {'name': 'name'},
+            {'name': ''},
+            {'name': False},
+        ])
+
+        self.assertListEqual(EmptyChar.search([('name', '=', 'name')]).mapped('name'), ['name'])
+        self.assertListEqual(EmptyChar.search([('name', '!=', 'name')]).mapped('name'), ['', False])
+        self.assertListEqual(EmptyChar.search([('name', 'ilike', 'name')]).mapped('name'), ['name'])
+        self.assertListEqual(EmptyChar.search([('name', 'not ilike', 'name')]).mapped('name'), ['', False])
+
+        self.assertListEqual(EmptyChar.search([('name', '=', '')]).mapped('name'), [''])
+        self.assertListEqual(EmptyChar.search([('name', '!=', '')]).mapped('name'), ['name'])
+        self.assertListEqual(EmptyChar.search([('name', 'ilike', '')]).mapped('name'), ['name', '', False])
+        self.assertListEqual(EmptyChar.search([('name', 'not ilike', '')]).mapped('name'), [False])
+
+        self.assertListEqual(EmptyChar.search([('name', '=', False)]).mapped('name'), [False])
+        self.assertListEqual(EmptyChar.search([('name', '!=', False)]).mapped('name'), ['name', ''])
+        self.assertListEqual(EmptyChar.search([('name', 'ilike', False)]).mapped('name'), ['name', '', False])
+        self.assertListEqual(EmptyChar.search([('name', 'not ilike', False)]).mapped('name'), [False])
```

Comment:
@rco-odoo Could you recheck the search behavior for non-translated fields?
It is hard for me to summarize if `False` records will be returned when search with `''` or `False`

Reply:
None of them are actually intuitive.
It all depends on how you interpret `False`.
I would say: take the existing behavior (before jsonified translations) and try to not change it.